mixes/reducer.js

```diff
@@ -62,6 +67,31 @@ function createReducer (config) {
     [loadMixEnd]: (state, action) => ({
       ...state, isLoading: false
     }),
+    [saveMix]: (state, action) => loop({
+      ...state, isLoading: true
+    }, Effects.batch([
+
+      // TODO: currently, the component passes a nestedMix into this saveMix action.payload
+      //       is this an antipattern? should i instead pass mixId to saveMix,
+      //       then in the reducer regenerate the flat mix, then call nestMix?
+      Effects.promise(runSaveMix, action.payload),
+
+      // TODO: if going with this approach, need to somehow get channels, clips
+      // Effects.promise(runSaveMix, nestMix({
+      //   ...action.payload,
+      //   channels: [],
+      //   clips: []
+      // })),
```

Comment:
your comments are bang on, yet i think for now it's best to keep passing in the pre-nested mix. i kept this (and use cases like this) in mind when [designing my own front-end framework](https://github.com/ahdinosaur/inux/issues/8#issuecomment-267492008), it'd be nice if each reducer was independently scope-able.